contrib/src/main/java/com/datatorrent/contrib/enrich/JDBCLoader.java

```diff
@@ -0,0 +1,201 @@
+      ArrayList<Object> keys = (ArrayList<Object>)key;
+      for (int i = 0; i < keys.size(); i++) {
+        getStatement.setObject(i + 1, keys.get(i));
+      }
+      return getStatement.executeQuery();
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected ArrayList<Object> getDataFrmResult(Object result) throws RuntimeException
+  {
+    try {
+      ResultSet resultSet = (ResultSet)result;
+      if (resultSet.next()) {
+        ResultSetMetaData rsdata = resultSet.getMetaData();
+        // If the includefields is empty, populate it from ResultSetMetaData
+        if (CollectionUtils.isEmpty(includeFieldInfo)) {
+          if (includeFieldInfo == null) {
+            includeFieldInfo = new ArrayList<>();
+          }
+          for (int i = 1; i <= rsdata.getColumnCount(); i++) {
+            String columnName = rsdata.getColumnName(i);
+            // TODO: Take care of type conversion.
+            includeFieldInfo.add(new FieldInfo(columnName, columnName, FieldInfo.SupportType.OBJECT));
+          }
+        }
+
+        ArrayList<Object> res = new ArrayList<Object>();
+        for (FieldInfo f : includeFieldInfo) {
+          res.add(getConvertedData(resultSet.getObject(f.getColumnName()), f));
+        }
+        return res;
+      } else {
+        return null;
+      }
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Object getConvertedData(Object object, FieldInfo f)
+  {
+    if (f.getType().getJavaType() == object.getClass()) {
+      return object;
+    } else {
+      logger.warn("Type mismatch seen for field {}, returning as it is", f.getColumnName());
+      return null;
+    }
+  }
+
+  private String generateQueryStmt()
+  {
+    String stmt = "select * from " + tableName + " where ";
+    boolean first = true;
+    for (FieldInfo fieldInfo : lookupFieldInfo) {
+      if (first) {
+        first = false;
+      } else {
+        stmt += " and ";
+      }
+      stmt += fieldInfo.getColumnName() + " = ?";
+    }
+
+    logger.info("generateQueryStmt: {}", stmt);
+    return stmt;
+  }
+
+  public String getQueryStmt()
+  {
+    return queryStmt;
+  }
+
+  /**
+   * Set the sql Prepared Statement if the enrichment mechanism is query based.
+   */
+  public void setQueryStmt(String queryStmt)
+  {
+    this.queryStmt = queryStmt;
+  }
+
+  public String getTableName()
+  {
+    return tableName;
+  }
+
+  /**
+   * Set the table name.
+   */
+  public void setTableName(String tableName)
+  {
+    this.tableName = tableName;
+  }
+
+  @Override
+  public void setFieldInfo(List<FieldInfo> lookupFieldInfo, List<FieldInfo> includeFieldInfo)
+  {
+    this.lookupFieldInfo = lookupFieldInfo;
+    this.includeFieldInfo = includeFieldInfo;
+    if (queryStmt == null) {
+      queryStmt = generateQueryStmt();
+    }
+  }
+
+  @Override
+  public Map<Object, Object> loadInitialData()
+  {
+    return null;
```

Comment:
I'm not sure if this is the best use of loadInitialData here.
If one wants, this loader can be extended to override this method.